zerver/tests/test_embedded_bot_system.py

```diff
@@ -83,11 +83,8 @@ def test_message_embedded_bot_with_invalid_service(self) -> None:
         service_profile = get_service_profile(bot_profile.id, 'helloworld')
         service_profile.name = 'invalid'
         service_profile.save()
-        with patch('logging.error') as logging_error_mock:
+        with self.assertLogs(level='ERROR') as m:
             self.send_stream_message(user_profile, "Denmark",
                                      content=f"@**{bot_profile.full_name}** foo",
                                      topic_name="bar")
-            logging_error_mock.assert_called_once_with(
-                "Error: User %s has bot with invalid embedded bot service %s",
-                bot_profile.id, "invalid",
-            )
+            self.assertRegexpMatches(m.output[0], r'ERROR:root:Error: User [0-9]* has bot with invalid embedded bot service invalid')
```

Comment:
I think basically all of the uses of `assertRegexpMatches` could be better written with an f-string on the to substitute the actual `user_id` (which is often known) and a check for equality.  Can you add another commit attempting to do that cleanup?

Reply:
I replaced all the instances of `assertRegexpMatches` in test_stripe.py` and `test_embedded_bot_system.py` to use `assertEqual`. However it is still used in `test_import_export.py`, as I used it there to avoid asserting complete file paths which changes depending on environment/system on which tests are run also we can migrate it later along with changes suggested for `check_logs` function.